extensions/blocks/instagram-gallery/edit.js

```diff
@@ -146,11 +152,27 @@ const InstagramGalleryEdit = props => {
 					label={ __( 'Instagram Gallery', 'jetpack' ) }
 					notices={ noticeUI }
 				>
-					<Button disabled={ isConnecting } isLarge isPrimary onClick={ connectToService }>
+					<Button
+						disabled={ ! isCurrentUserConnected || isConnecting }
+						isLarge
+						isPrimary
+						onClick={ connectToService }
+					>
 						{ isConnecting
 							? __( 'Connecting…', 'jetpack' )
 							: __( 'Connect your Instagram account', 'jetpack' ) }
 					</Button>
+
+					<Notice isDismissible={ false } status="info">
+						{ __(
+							'To connect your Instagram account, you need to link your account to WordPress.com.',
+							'jetpack'
+						) }
+						<br />
+						<Button href={ addQueryArgs( 'admin.php', { page: 'jetpack#/dashboard' } ) } isLink>
+							{ __( 'Open the Jetpack dashboard', 'jetpack' ) }
+						</Button>
+					</Notice>
```

Comment:
Do we want to only display the notice when you're not connected to WordPress.com yet?

Reply:
Yup!
Sorry about this, I realized way too late that the `{ isCurrentUserConnected && <Notice> }` got left out from the commit, and remained in the uncommitted follow-ups!